datamimic_ce/tasks/generate_task.py

```diff
@@ -41,39 +43,12 @@
 from datamimic_ce.utils.in_memory_cache_util import InMemoryCache
 from datamimic_ce.utils.multiprocessing_page_info import MultiprocessingPageInfo
 
+# TODO: set dynamic debug mode
+ray.init(ignore_reinit_error=True, local_mode=True)
 
-def _wrapper(args):
-    """
-    Wrapper multiprocessing function to deserialize args and execute the generate function.
 
-    :param args: Tuple containing necessary arguments.
-    :return: Result from single_process_execute_function.
-    """
-    (
-        local_ctx,
-        statement,
-        chunk_data,
-        single_process_execute_function,
-        namespace_functions,
-        mp_idx,
-        page_size,
-        mp_chunk_size,
-    ) = args
-    # Re-initialize logging for this child process
-    loglevel = os.getenv("LOG_LEVEL", "INFO")
-    setup_logger_func = local_ctx.class_factory_util.get_setup_logger_func()
-    app_settings = local_ctx.class_factory_util.get_app_settings()
-    setup_logger_func(logger_name=app_settings.DEFAULT_LOGGER, task_id=local_ctx.task_id, level=loglevel)
-
-    # Deserialize utility functions
-    namespace_functions = dill.loads(namespace_functions)
-    local_ctx.namespace.update(namespace_functions)
-    local_ctx.generators = dill.loads(local_ctx.generators)
-
-    return single_process_execute_function((local_ctx, statement, chunk_data, mp_idx, page_size, mp_chunk_size))
-
-
-def _geniter_single_process_generate(args: tuple) -> dict[str, list]:
+def _geniter_single_process_generate(context: SetupContext | GenIterContext, stmt: GenerateStatement, page_start: int,
```

Comment:
should not exists anymore ... and we should re-evaluate the naming of the function to make it more clear what they are actually doing